test/statements/manage/ContextManagerSyntax.chpl

```diff
@@ -0,0 +1,199 @@
+//
+// TODO: Check to make sure 'const' storage kind is immutable.
+//
+
+use TrackingRecord;
+
+proc r.enterThis() {
+  writeln('entering');
+}
+
+proc r.leaveThis(in err: owned Error?) {
+  writeln('leaving');
+  if err then try! { throw err; }
+}
+
+record res {
+  proc doSomething() {
+    writeln('doing something with resource');
+  }
+}
+
+var globalRes = new res();
+
+record man {
+  var x = new r();
+
+  proc enterThis() ref {
+    writeln('proc man.enterThis() ref: res');
+    return globalRes;
+  }
+
+  proc enterThis() const ref {
+    writeln('proc man.enterThis() const ref: res');
+    return globalRes;
+  }
+
+  proc enterThis() {
+    writeln('proc man.enterThis(): res');
+    return globalRes;
+  }
+
+  proc leaveThis(in err: owned Error?) {
+    writeln('leaving');
+    if err then try! { throw err; }
+  }
+}
+
+proc doSomething() {
+  writeln('doing something');
+}
+
+proc test1() {
+  writeln('T1');
+  // Calls the value overload of 'enterThis'.
+  manage new man() do doSomething();
+}
+test1(); writeln();
+
+proc test2() {
+  writeln('T2');
+  // Calls the value overload of 'enterThis'.
+  manage new man() {
+    doSomething();
+  }
+}
+test2(); writeln();
+
+// Make sure 'manager' stays alive till end of block.
+proc test3() {
+  writeln('T3');
+  var myManager = new man();
+  manage myManager do doSomething();
+}
+test3(); writeln();
+
+// Make sure 'manager' stays alive till end of block, nested.
+proc test4() {
+  writeln('T4');
+  var myManager = new man();
+  manage myManager, new man() {
+    doSomething();
+  }
+}
+test4(); writeln();
+
+// Make sure both managers stay alive till end of block.
+proc test5() {
+  writeln('T5');
+  var myManager1 = new man();
+  var myManager2 = new man();
+  manage myManager1, myManager2 do
+    doSomething();
+}
+test5(); writeln();
+
+// TODO: This may be dropped if we decide we don't want explicit storage.
```

Comment:
I wasn't following the design discussion towards the end very carefully, and didn't know that we haven't decided on the explicit storage design yet. So, if things are still in flux, I think we shouldn't merge this with the support for explicit storage.

I don't necessarily suggest removing all the code for it, but you can just drop the relevant rule from the parser, or rewire it to some more meaningful error message.

Reply:
I don't think we're set on implicit storage yet either. But either way, I think it's fine to leave the syntax as is, as the manage statement generates `--warn-unstable` warnings, which gives us the freedom to change the syntax next release.